services/auth/oauth2.go

```diff
@@ -59,31 +59,32 @@ func (o *OAuth2) Name() string {
 	return "oauth2"
 }
 
-// userIDFromToken returns the user id corresponding to the OAuth token.
-// It will set 'IsApiToken' to true if the token is an API token and
-// set 'ApiTokenScope' to the scope of the access token
-func (o *OAuth2) userIDFromToken(req *http.Request, store DataStore) int64 {
+// parseToken returns the token from request, and a boolean value
+// representing whether the token exists or not
+func (o *OAuth2) parseToken(req *http.Request) (string, bool) {
```

Comment:
Looks like it could become a standalone function but not a method of OAuth2?

Reply:
I think so too, but I don't know where to put this function in the project. It should be a utility function ? Or just remove the receiver ?

Reply:
later.

Reply:
I got it !